plugins/rocksdb/src/main/java/org/hyperledger/besu/plugin/services/storage/rocksdb/segmented/RocksDBColumnarKeyValueStorage.java

```diff
@@ -68,48 +66,51 @@
 import org.slf4j.LoggerFactory;
 
 /** The RocksDb columnar key value storage. */
-public class RocksDBColumnarKeyValueStorage
+public abstract class RocksDBColumnarKeyValueStorage
     implements SegmentedKeyValueStorage<RocksDbSegmentIdentifier> {
 
   private static final Logger LOG = LoggerFactory.getLogger(RocksDBColumnarKeyValueStorage.class);
-  private static final String DEFAULT_COLUMN = "default";
+  static final String DEFAULT_COLUMN = "default";
   private static final String NO_SPACE_LEFT_ON_DEVICE = "No space left on device";
   private static final int ROCKSDB_FORMAT_VERSION = 5;
   private static final long ROCKSDB_BLOCK_SIZE = 32768;
-  private static final long ROCKSDB_BLOCKCACHE_SIZE_HIGH_SPEC = 1_073_741_824L;
-  private static final long ROCKSDB_MEMTABLE_SIZE_HIGH_SPEC = 1_073_741_824L;
+  /** RocksDb blockcache size when using the high spec option */
+  protected static final long ROCKSDB_BLOCKCACHE_SIZE_HIGH_SPEC = 1_073_741_824L;
+  /** RocksDb memtable size when using the high spec option */
+  protected static final long ROCKSDB_MEMTABLE_SIZE_HIGH_SPEC = 1_073_741_824L;
 
   static {
     RocksDbUtil.loadNativeLibrary();
   }
 
-  private final DBOptions options;
-  private final TransactionDBOptions txOptions;
-  private final OptimisticTransactionDB db;
   private final AtomicBoolean closed = new AtomicBoolean(false);
-  private final Map<String, RocksDbSegmentIdentifier> columnHandlesByName;
-  private final RocksDBMetrics metrics;
   private final WriteOptions tryDeleteOptions =
       new WriteOptions().setNoSlowdown(true).setIgnoreMissingColumnFamilies(true);
   private final ReadOptions readOptions = new ReadOptions().setVerifyChecksums(false);
-
-  /**
-   * Instantiates a new RocksDb columnar key value storage.
-   *
-   * @param configuration the configuration
-   * @param segments the segments
-   * @param metricsSystem the metrics system
-   * @param rocksDBMetricsFactory the RocksDb metrics factory
-   * @throws StorageException the storage exception
-   */
-  public RocksDBColumnarKeyValueStorage(
-      final RocksDBConfiguration configuration,
-      final List<SegmentIdentifier> segments,
-      final MetricsSystem metricsSystem,
-      final RocksDBMetricsFactory rocksDBMetricsFactory)
-      throws StorageException {
-    this(configuration, segments, List.of(), metricsSystem, rocksDBMetricsFactory);
-  }
+  private final MetricsSystem metricsSystem;
+  private final RocksDBMetricsFactory rocksDBMetricsFactory;
+  private final RocksDBConfiguration configuration;
+  private Map<Bytes, String> segmentsById;
+  /** RocksDB DB options */
```

Comment:
nit: there are a lot more comments here now, but maybe it's just me but I don't think they add much value in this case

Reply:
Added to comply with the javadoc rules